eldritch/encounters.py

```diff
@@ -252,11 +252,134 @@ def Graveyard6(char):
 def Graveyard7(char):
   return events.Nothing()
 
+def Society1(char):
+  move = events.ForceMovement(char, "Woods")
+  encounter = events.Encounter(char, "Woods", 2)
+  accept = events.Sequence([move, encounter], char)
+  return events.BinaryChoice(char, "Accept Ride?", "Yes", "No", accept, events.Nothing())
+def Society2(char):
+  prereq = events.AttributePrerequisite(char, "dollars", 3, "at least")
+  move = events.ForceMovement(char, "Southside")
+  luck = events.Check(char, "luck", -1)
+  move_dream = events.ForceMovement(char, "Dreamlands1")
+  # TODO: don't hard-code.
+  enc = events.GateEncounter(char, "Dreamlands", {"blue", "green", "red", "yellow"})
+  ret = events.ForceMovement(char, "Society")
+  dreamlands =  events.Sequence([move_dream, enc, ret], char)
+  spell = events.Draw(char, "spells", 1)
+  pay = events.Loss(char, {"dollars": 3})
+  searchstacks = events.PassFail(char, luck, spell, dreamlands)
+  library = events.Sequence([pay, searchstacks], char)
+  choice = events.BinaryChoice(char,"Pay $3 to access the private library", "Yes", "No", library, move)
+  return events.PassFail(char, prereq, choice, move)
+def Society3(char):
+  check = events.Check(char, "sneak", -1)
+  sanity = events.Gain(char, {"sanity": 1})
+  curse = events.Curse(char)
+  stamina = events.Loss(char, {"stamina": 2})
+  move = events.ForceMovement(char, "Southside")
+  return events.PassFail(char, check, sanity, events.Sequence([curse, stamina, move], char))
 def Society4(char):
   check = events.Check(char, "luck", -1)
   skill = events.Sequence([events.Draw(char, "skills", 1), events.Delayed(char)], char)
   cond = events.Conditional(char, check, "successes", {0: events.Nothing(), 2: skill})
   return events.Sequence([check, cond], char)
+def Society5(char):
+  return events.Loss(char, {"sanity": 1})
+def Society6(char):
+  # TODO: spend gate trophy
+  prereq = events.AttributePrerequisite(char, "dollars", 1, "at least") # TODO: actually a gate trophy
+  # TODO: prerquisite on ally
+  ally = events.Draw(char, "allies", "Old Professor")
```

Comment:
Can now be replaced with GainAllyOrReward.

Reply:
implemented this except for 1 dollar instead of gate trophy